### DISCUSSION TITLE
How to use L2 Loss , L1 Loss or MSE Loss while training Deepar Model ?

### DISCUSSION THREAD

**Discussion by sayedathar11** [awslabs/gluonts#1810] [General]
Hiii ,
Our team is using GluonTS Deepar model for time series forecasting our target has range 0 to 1 , As per our information , GluonTS Deepar uses StudentTOutput and it doesn't uses L1 Loss , L2 Loss or Mse Loss , we wanted to ask how to change loss function in GluonTS Deepar ? Is there any fucntionality that allows us to select L1 Loss or L2 Loss or Mse Loss or we can only go with StudentTOutput ?
@mbohlkeschneider @lostella  can you please help with it ?

**Comment by hardik-raja-dsw** [comment]
hi, we had the same query related to loss function, any help on same would be appreciated, Thanks in advance

**Comment by mbohlkeschneider** [comment]
Hi @sayedathar11  and @hardik-raja-dsw, 

The Laplace and Gaussian distribution essentially reduces to L1 and L2 loss when fixing the variance, respectively. For L1/Laplace, this is done here: https://github.com/awslabs/gluon-ts/blob/master/src/gluonts/mx/distribution/laplace.py#L123. 

You can implement something analogous for the `Gaussian` distribution to get an L2 loss. 

Hope that helps.

**Reply by sayedathar11** [reply]
@mbohlkeschneider That works , Thanks alot for your reply
